src/html/scan.js

```diff
@@ -514,7 +514,28 @@ function fileDragHover(e) {
 function fileSelectHandler(e) {
   fileDragHover(e);
   const files = e.target.files || e.dataTransfer.files;
-  uploadFile(files[0]);
+  const fileExt = files[0].name.split('.').pop();
```

Comment:
I have never seen this way of pulling an extension off a filename, _and I flippin' **love** it_.

Reply:
can't claim full credit on this one.. :)